src/main/java/com/github/manolo8/darkbot/core/manager/PetManager.java

```diff
@@ -736,6 +736,21 @@ public void setModule(long gearsSprite) {
             Main.API.callMethodChecked(true, "23(hide)(26)008211400",
                     152, Main.API.readLong(gearsSprite, 176));
         }
+
+        @Override
+        public String getName() {
+            return name;
+        }
+
+        @Override
+        public String getFuzzyName() {
+            return fuzzyName;
+        }
```

Comment:
For performance reasons, we may want to just make gear have a getNpcInfo, that gets mapped and updated by the bot (when updating gears, if name changes.

That way we avoid N^2 where each NpcInfo iterates thru all LocatorPick to find the matching one being ran every time, and in several places.

Reply:
I thought about it but not always gear can find the corresponding NpcInfo, so have to think of a way to 'fix' it, when the bot see the given npc for the first time

Reply:
You can make the result nullable, at the end of the day the issue isn't going to be much different to what we have now, if no npc info matches the gear then it just can't really be selected